src/renderer/tvgWgCanvas.cpp

```diff
@@ -50,11 +50,13 @@ WgCanvas::WgCanvas() : Canvas(nullptr), pImpl(nullptr)
 
 WgCanvas::~WgCanvas()
 {
-    delete pImpl;
+    #ifdef THORVG_WG_RASTER_SUPPORT
+    static_cast<WgRenderer*>(Canvas::pImpl->renderer)->surface = nullptr;
```

Comment:
@SergeyLebedkin  Does this really necessary job?  it shouldn't be like this.

Reply:
@hermet Yes. We need some how tell to renderer that surface already destroed. If not, it will raise an exception because of Canvas::Impl design: it calls renderer->sync() in disctuctor, but surface is invalid in this stage

```
struct Canvas::Impl
{
    ~Impl()
    {
        //make it sure any deferred jobs
        renderer->sync();

        clearPaints();

        if (renderer->unref() == 0) delete(renderer);
    }

```

Reply:
@SergeyLebedkin user should not destroy a surface, assigned in wg canvas. or they should use target() to unassign before destroying surface.

Reply:
@hermet I think that we can use terget for un-usigning. Will fix now